src/components/dls/Modal/Modal.stories.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState } from 'react';
+
+import Modal from './Modal';
+
+export default {
+  title: 'dls|Modals',
+  component: Modal,
+  argTypes: {
+    visible: {
+      defaultValue: false,
+      description: `Whether the modal dialog is visible or not`,
+      table: {
+        category: 'Required',
+      },
+    },
+    centered: {
+      defaultValue: false,
+      description: `Whether the modal should be centered or not.`,
+      table: {
+        category: 'Optional',
+      },
+      options: [true, false],
+      control: { type: 'select' },
+    },
+    showCloseIcon: {
+      defaultValue: true,
+      description: `Whether a close (x) button is visible on the top right of the modal or not.`,
+      table: {
+        category: 'Optional',
+      },
+      options: [true, false],
+      control: { type: 'select' },
+    },
+    canCloseByKeyboard: {
+      defaultValue: true,
+      description: `Whether to support pressing esc to close the modal or not.`,
+      table: {
+        category: 'Optional',
+      },
+      options: [true, false],
+      control: { type: 'select' },
+    },
+    closeWhenClickingOutside: {
+      defaultValue: true,
+      description: `Whether to close the modal dialog when the mask (area outside the modal) is clicked.`,
+      table: {
+        category: 'Optional',
+      },
+      options: [true, false],
+      control: { type: 'select' },
+    },
+    children: {
+      description: 'The modal body',
+      table: {
+        category: 'Optional',
+      },
+    },
+    title: {
+      description: 'The title of the modal.',
+      table: {
+        category: 'Optional',
+      },
+    },
+    width: {
```

Comment:
ideally we don't specify a width. We instead have a `size` prop with preset widths like: `default` `large`.

This constraint is good for consistency

Reply:
I thought having the width in pixels would give a lot of flexibility when using the modal versus when we have pre-defined limited values because they might not match exactly what the user of the `Modal` is looking for. I was looking at the [antd's component API](https://ant.design/components/modal/#API) for modal and they got it defined in pixels. 

<img width="910" alt="Screen Shot 2021-07-25 at 11 00 22" src="https://user-images.githubusercontent.com/15169499/126887277-25f18613-6dd3-4cdf-8f82-4f493915943a.png">